app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -13,11 +13,20 @@ def stop_event_monitor
   # #stop_event_monitor is called.
   def monitor_events
     _log.info "#{log_prefix} Monitoring for events"
-    event_monitor_handle.each_batch do |events|
-      event_monitor_running
-      _log.debug "#{log_prefix} Received events #{events.collect { |e| parse_event_type(e) }}"
-      @queue.enq events
-      sleep_poll_normal
+    begin
+      event_monitor_handle.each_batch do |events|
+        event_monitor_running
+        _log.debug "#{log_prefix} Received events #{events.collect { |e| parse_event_type(e) }}"
+        @queue.enq events
+        sleep_poll_normal
+      end
+    rescue ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream::TopicNotFound
+      warn = "Unable to find topic to subscribe for events. Please see the documentation on how to setup events for"\
+        " the Google Cloud Provider."
+
+      _log.warn("#{log_prefix} #{warn}")
+      # Sleep for 5 minutes, heartbeating every 30 seconds
+      sleep_and_heartbeat(30.seconds, 5.minutes)
```

Comment:
As I said in my main comment, I'd just raise `ProviderUnreachable` with a descriptive message and remove this timeout.

Although I think its nice to have, but it should be rather handled in the base runner/catcher and not on a provider level.